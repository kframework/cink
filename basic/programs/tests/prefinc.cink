/*
  This example exhibits different uses of the prefixed increment operator.
*/
#include <iostream>
using namespace std;

int main() {
  int x = 1,y, z;
  y = ++x + 3;
  cout << "x = " << x << "; y = " << y << endl;
  return 0;
} 
