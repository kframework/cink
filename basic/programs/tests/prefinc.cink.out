x = 2; y = 5
<T>
    <sideEffects>
        .K
    </sideEffects>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <env>
            main |-> #symLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symLoc(1)
    </genv>
    <store>
        #symLoc(1) |-> lambda ( .Decls @ ((((((((int x) ;) ((x = 1) ;)) ((
          int y) ;)) ((int z) ;)) ((y = ((++ x) + 3)) ;)) ((((((cout << "x = "
          ) << x) << "; y = ") << y) << endl) ;)) (return 0 ;)) )
        #symLoc(2) |-> 2
        #symLoc(3) |-> 5
        #symLoc(4) |-> 0
    </store>
</T>
